src/UsesResolver.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+/**
+ * Laravel IDE Helper Generator
+ *
+ * @author    Barry vd. Heuvel <barryvdh@gmail.com>
+ * @copyright 2014 Barry vd. Heuvel / Fruitcake Studio (http://www.fruitcakestudio.nl)
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      https://github.com/barryvdh/laravel-ide-helper
+ */
+
+namespace Barryvdh\LaravelIdeHelper;
+
+use Illuminate\Support\Arr;
+use PhpParser\Node\Stmt\GroupUse;
+use PhpParser\Node\Stmt\Namespace_;
+use PhpParser\Node\Stmt\Use_;
+use PhpParser\Node\Stmt\UseUse;
+use PhpParser\ParserFactory;
+
+class UsesResolver
+{
+    /**
+     * @var array
+     */
+    protected $classAliases = [];
```

Comment:
From a design perspective of this class, I suggest to make it stateless as there's no benefit to it.

Suggest approach:
- get rid of this property
- just return the data array from `loadFromCode`
- get rid of `addClassAlias
- change all the `load*` methods to just return the data you return from `loadFromCode`
- which means the call in `\Barryvdh\LaravelIdeHelper\Alias::__construct` is just `(new UsesResolver())->loadFromClass($this->root);`
- adapt test

Most classes in this library have their fair share of states, which is nightmare to trace properly.